src/circuits/mimc/round.tcc

```diff
@@ -0,0 +1,72 @@
+// DISCLAIMER:
+// Content taken and adapted from:
+// https://github.com/HarryR/ethsnarks/blob/master/src/gadgets/mimc.hpp
+
+#ifndef __ZETH_MIMC_ROUND_TCC__
+#define __ZETH_MIMC_ROUND_TCC__
+
+namespace libzeth {
+
+template<typename FieldT>
+MiMCe7_round_gadget<FieldT>::MiMCe7_round_gadget(
+        libsnark::protoboard<FieldT>& pb,
+        const libsnark::pb_variable<FieldT> in_x,
+        const libsnark::pb_variable<FieldT> in_k,
+        const FieldT& in_C,
+        const bool in_add_k_to_result,
+        const std::string &annotation_prefix
+    ) :
+        libsnark::gadget<FieldT>(pb, annotation_prefix),
+        x(in_x), k(in_k), C(in_C),
+        add_k_to_result(in_add_k_to_result)
+    {
+      a.allocate(pb, FMT(annotation_prefix, ".a"));
+      b.allocate(pb, FMT(annotation_prefix, ".b"));
+      c.allocate(pb, FMT(annotation_prefix, ".c"));
+      d.allocate(pb, FMT(annotation_prefix, ".d"));
+     }
+
+template<typename FieldT>
+const libsnark::pb_variable<FieldT>& MiMCe7_round_gadget<FieldT>:: result() const {
+        return d;
+    }
+
+template<typename FieldT>
+void MiMCe7_round_gadget<FieldT>::generate_r1cs_constraints() {
+        libsnark::linear_combination<FieldT> t = x + k + C; // define `t` as the variable to exponentiate
+
+        this->pb.add_r1cs_constraint(libsnark::r1cs_constraint<FieldT>(t, t, a), ".a = t*t"); // Add constraint `a = t^2`
+        this->pb.add_r1cs_constraint(libsnark::r1cs_constraint<FieldT>(a, a, b), ".b = a*a"); // Add constraint `b = a^2 = t^4`
```

Comment:
In the rest of the code base of `zeth` the convention to write annotation of constraint was:
```
this->pb.add_r1cs_constraint(r1cs_constraint<FieldT>(
          foo,
          bar,
          foobar,
     ),
     FMT(this->annotation_prefix, " constraint_annotation")
);
```
However here annotations are prefixed with a `.` and not suffixed by the annotation of the gadget

Reply:
Fixed